src/main/java/cms/gov/madie/measure/services/VersionService.java

```diff
@@ -405,4 +388,24 @@ private boolean checkCaseNumberExists(List<TestCase> testCases) {
     }
     return true;
   }
+
+  List<TestCase> assignCaseNumbers(
+      List<TestCase> testCases, String measureId, boolean isFlagEnabled, boolean caseNumberExists) {
+    return testCases.stream()
+        .sorted(
+            Comparator.comparing(
+                TestCase::getCreatedAt, Comparator.nullsFirst(Comparator.naturalOrder())))
+        .map(
+            testCase -> {
+              if (isFlagEnabled) {
+                if (caseNumberExists) {
+                  sequenceService.generateSequence(measureId);
```

Comment:
There will be an issue with this approach, We are assuming that the versioned measures will have sequential case Id but that might not be the case, If the Test Case in Versioned measure has case number 1,2,3,5,6,9. Then according to this logic it is assuming that the first 6 test cases will be 1 - 6 and hence it can create duplicates

Reply:
@RohitKandimalla This is addressed, thanks for the tip.